src/mysql_protocol.erl

```diff
@@ -846,6 +849,12 @@ flooring(Value) ->
         Trunc > Value -> Trunc - 1 %% for negative values
     end.
 
+%% @doc This function is designed to avoid the bug caused by delayed sub binary
+%% creation optimization. This bug was found in OTP 21.0 and was fixed in OTP
+%% 21.1. Ref: https://github.com/erlang/otp/pull/1911
+fetch_length_and_data(<<Length, Data/binary>>) ->
+    {Length, Data}.
```

Comment:
I don't like the name 'fetch'. Fetch means get from some storage or network. We're just treating the argument to the function. How about `parse_length_prefixed_data`?

Reply:
The existing function `lenenc_int/1` does this. It can be used instead of this function.

The documentation you added here can be put in the commit message. I think it's enough, as long as we have automatic tests for 21.0.